pkg/tempdb/factory.go

```diff
@@ -185,12 +202,18 @@ func (o *onInstanceFactory) Create(ctx context.Context) (sql *sql.DB, dropper Dr
 		INSERT INTO %s DEFAULT VALUES;
 	`, sanitizedSchemaName, sanitizedTableName, sanitizedTableName)
 	if _, err := tempDbConn.ExecContext(ctx, createMetadataStmts); err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	return tempDbConn, func(ctx context.Context) error {
-		_ = tempDbConn.Close()
-		return o.dropTempDatabase(ctx, tempDbName)
+	return &Database{
+		ConnPool: tempDbConn,
+		ExcludeMetadatOptions: []schema.GetSchemaOptions{
+			schema.WithExcludeSchemas(o.options.metadataSchema),
```

Comment:
This is the core of the PR. Without this, all acceptance tests would fail because they would try to diff out the metadata table.